lib/rules/no-constant-condition.js

```diff
@@ -90,14 +90,25 @@ module.exports = {
          * @private
          */
         function isConstant(node, inBooleanPosition) {
+
+            // node.elements can return null values in the case of sparse arrays ex. [,]
+            if (!node) {
+                return true;
+            }
             switch (node.type) {
                 case "Literal":
                 case "ArrowFunctionExpression":
                 case "FunctionExpression":
                 case "ObjectExpression":
-                case "ArrayExpression":
                     return true;
 
+                case "ArrayExpression": {
+                    if (!(node.parent.operator === "+")) {
+                        return true;
+                    }
+                    return node.elements.every(element => isConstant(element, false));
```

Comment:
Perhaps we should also check the parent's type and continue only if it's `BinaryExpression`? I think that this at the moment also allows unary `+` which almost always evaluates to `NaN` (except when the array has 1 element, but then the array is unnecessary anyway).

I was also thinking to allow `+` on an array only if the other operand is a string literal, to minimize the change in behavior, but I'm not sure about that.